cli/server.go

```diff
@@ -2363,10 +2363,8 @@ func ConnectToPostgres(ctx context.Context, logger slog.Logger, driver string, d
 	if !version.Next() {
 		// it's critical we assign to the err variable, otherwise the defer statement
 		// that runs db.Close() will not execute it
-		if err = version.Err(); err != nil {
-			return nil, xerrors.Errorf("no rows returned for version select: %w", err)
-		}
-		return nil, xerrors.Errorf("no rows returned for version select")
+		err = xerrors.Errorf("no rows returned for version select: %w", version.Err())
+		return nil, err
```

Comment:
Can we just name this error something else? Like `dbError`?

I say that because a few lines up is `version, err := sqlDB.QueryContext(ctx, "SHOW server_version_num;")`. Does that not shadow the `err` variable and break this assignment to the correct `err` here?